lib/open_project/text_formatting/matchers/attribute_macros.rb

```diff
@@ -50,14 +52,29 @@ def self.applicable?(content)
         content.include?("Label:") || content.include?("Value:")
       end
 
-      def self.process_match(m, _matched_string, _context)
+      def self.work_package_context?(context)
+        context[:object].is_a?(WorkPackage) || context[:object].is_a?(API::V3::WorkPackages::WorkPackageEagerLoadingWrapper)
+      end
+
+      def self.relative_work_package_embed?(macro_attributes)
+        macro_attributes[:id].nil? && macro_attributes[:model] == "workPackage"
+      end
+
+      def self.process_match(match, _matched_string, context)
         # Leading string before match
         macro_attributes = {
-          model: m[1],
-          id: m[4] || m[3],
-          attribute: m[6] || m[5]
+          model: match[1],
+          id: match[4] || match[3],
+          attribute: match[6] || match[5]
         }
-        type = m[2].downcase
+        type = match[2].downcase
+
+        # If no ID is given, try to use the current work package ID if applicable
+        #  workPackageValue can be used in e.g. wiki and meeting notes without a work package,
+        #  relative embedding is not supported in these cases
+        if relative_work_package_embed?(macro_attributes) && work_package_context?(context)
```

Comment:
I'd like to see `relative_embed` to work for both projects and work packages in the backend. While I understand why you try and resolve it in the backend, I'd want to see both models have the same lookup consistently. It will be hard to extend if one is magically looked up in the frontend, and the other is not.

This will allow us to make it easier to parse in the frontend, as you'd always expect an ID present on the attribute macro.

Reply:
Good point. Will do!